cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+# Radius AWS Workload Identity Support
+
+* **Status**: In Review
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+A software workload such as a container-based application, service or script needs an identity to authenticate, access, and communicate with services that are distributed across different platforms and/or cloud providers. Radius uses AWS IAM (Identity and Access Management) credentials to deploy and access AWS resources today:
+ 
+```
+rad credential register aws --access-key-id <access-key-id> --secret-access-key <secret-access-key>
+```
+ 
+These IAM credentials, including the access key should be rotated regularly to reduce the chance of unauthorized access. A more secure option than using these credentials is to use IRSA.
+ 
+IRSA (IAM Roles for Service Accounts) is used for authenticating applications running within Kubernetes pods. When we use IRSA, we associate an IAM role directly with a Kubernetes service-account, allowing the pods associated with the service-account to assume that role. IRSA does not require us to configure access key and  secret. It relies identity tokens (OIDC tokens) to authenticate with AWS services. These tokens have a default expiration (configurable). However, they automatically refresh when needed, transparent to the user.
+
+The goal of the scenario is to enable infrastructure operators to configure IRSA support for the AWS provider in Radius to deploy and manage AWS resources.
+
+## Terms and definitions
+
+
+## Objectives
+
+> Issue Reference: https://github.com/radius-project/radius/issues/7618 
+
+### Goals
+
+* Radius users can configure AWS provider and enable Radius to use IRSA for authentication and deployment of application.
+* IRSA can be configured via interactive experience.
+* IRSA can be configured manually.
+* Radius users can deploy and manage AWS resources without having to manage AWS access key id and secret
+
+### Non-goals
+
+* Azure Managed Identity support
+* Azure Workload Identity support
+* Ability of radified applications to be able to use IRSA for authenticating AWS operations.
+
+### User scenarios
+
+As a user, I should be able to configure Radius with IRSA for managing AWS resources.
+
+## Design
+
+### High Level Design for IRSA
+
+During installation, Radius should allow user to enable IRSA for AWS while conifguring AWS provider. Then the user should be able to enter the roleARN associated with desired policies which will be stored as the AWS credential.
+
+At a high level, the full flow to setup IRSA looks as below:
+
+1. configure the cluster with oidc provider. have the role-arn configured with desired policies ready.
```

Comment:
Can we have 2 OIDC providers in the same cluster? Asking because what if we have resources on Azure and also on AWS?

Reply:
same OIDC should  provide the authentication needed for both aws and azure.